Assets/VRTK/Source/Scripts/Utilities/VRTK_SharedMethods.cs

```diff
@@ -472,6 +469,109 @@ public static Vector3 AxisDirection(int axisIndex, Transform givenTransform = nu
             return worldDirections[(int)Mathf.Clamp(axisIndex, 0f, worldDirections.Length)];
         }
 
+        /// <summary>
+        /// The AddListValue method attempts to add a value in the given list if the value does not already exist. `forceAdd = true` will always add a new value to the list even if it already exists.
+        /// </summary>
+        /// <typeparam name="TValue">The datatype for the list value.</typeparam>
+        /// <param name="list">The list to retrieve the value from.</param>
+        /// <param name="value">The value to attempt to add to the list.</param>
+        /// <param name="forceAdd">If this is `true` then the value provided will always be appended to the list. If this is `false` the value provided will only be added to the list if it doesn't already exist.</param>
+        /// <returns>Returns `true` if the given value was successfully added to the list. Returns `false` if the given value already existed in the list and `forceAdd = false`.</returns>
+        public static bool AddListValue<TValue>(ref List<TValue> list, TValue value, bool forceAdd = false)
```

Comment:
Why is `list` a `ref`? If you change this do a null check like in the dictionary method.

Reply:
Because you're passing a list that needs changing outside of the scope of the method.

Reply:
You are changing the list, but you aren't changing the reference AFAICS. I see no need to use `ref` here.

Reply:
i suppose list is already being passed as a reference because c# passes objects as references by default right?

Reply:
Exactly! You only need `ref` if you want to change the reference the caller has to a completely different list.